flow-typed/custom/progression-engine.js

```diff
@@ -0,0 +1,1219 @@
+      | {
+          payload: {
+            content:
+              | { ref: string, type: string, version?: string }
+              | { ref: string, type: 'video' | 'pdf', version?: string },
+            nextContent:
+              | { ref: string, type: string, version?: string }
+              | { ref: string, type: 'video' | 'pdf', version?: string }
+          },
+          type: 'extraLifeRefused'
+        }
+      | {
+          payload: {
+            content:
+              | { ref: string, type: string, version?: string }
+              | { ref: string, type: 'video' | 'pdf', version?: string },
+            resource: { ref: string, type: 'video' | 'pdf', version?: string }
+          },
+          type: 'resource'
+        }
+      | {
+          payload: {
+            instructions: Array<{
+              field: string,
+              type: 'add' | 'set',
+              value: number | boolean | string
+            }> | null,
+            nextContent:
+              | { ref: string, type: string, version?: string }
+              | { ref: string, type: 'video' | 'pdf', version?: string }
+          },
+          type: 'move'
+        }
+    >,
+    content:
+      | { ref: string, type: string, version?: string }
+      | { ref: string, type: 'video' | 'pdf', version?: string },
+    engine: { ref: string, version: string },
+    engineOptions: { lives?: number, livesDisabled?: boolean },
+    state?: {
+      content?:
+        | { ref: string, type: string, version?: string }
+        | { ref: string, type: 'video' | 'pdf', version?: string },
+      nextContent:
+        | { ref: string, type: string, version?: string }
+        | { ref: string, type: 'video' | 'pdf', version?: string }
+    } & {
+      allAnswers: Array<{
+        answer: Array<string>,
+        isCorrect: ?boolean,
+        slideRef: string
+      }>,
+      content?:
+        | { ref: string, type: string, version?: string }
+        | { ref: string, type: 'video' | 'pdf', version?: string },
+      hasViewedAResourceAtThisStep: boolean,
+      isCorrect: ?boolean,
+      lives: number,
+      livesDisabled?: boolean,
+      nextContent?:
+        | { ref: string, type: string, version?: string }
+        | { ref: string, type: 'video' | 'pdf', version?: string },
+      remainingLifeRequests: number,
+      requestedClues: Array<string>,
+      slides: Array<string>,
+      stars: number,
+      step: { current: number },
+      variables: { [_: string]: string | boolean | number },
+      viewedResources: Array<{
+        ref: string,
+        resources: Array<string>,
+        type: mixed /* UNEXPECTED TYPE: EvalT */
+      }>
+    }
+  }): {
+    answerBoundaryLimit: number,
+    lives: number,
+    livesDisabled: boolean,
+    maxTypos: number,
+    remainingLifeRequests: number,
+    slidesToComplete: number,
+    starsPerAskingClue: number,
+    starsPerCorrectAnswer: number,
+    starsPerResourceViewed: number,
+    version: string
+  };
+}
```

Comment:
pas trouvé comment éviter de passer par flow-typed :(

Reply:
you don't need to generate it, your other packages using `flow` will be in es6, so the packages imported will automatically covered by flow.

Reply:
This file should be added only with `flow-typed` cli, which installs definition for not typed packages hosted in [official repository](https://github.com/flow-typed/flow-typed/tree/master/definitions/npm).

Btw, this file will only be covered and read by flow in the monorepo, not by another project using this package.

Reply:
Look at this example, the guy is using a monorepo using lerna, and their packages are importing types from other packages, without any issue: 

https://github.com/wtgtybhertgeghgtwtg/redux-modules

Maybe you need to make this in your flowconfig:
https://github.com/wtgtybhertgeghgtwtg/redux-modules/blob/master/.flowconfig#L6